GitHub_Learning/Feature/Dashboard/Application/DashboardController.swift

```diff
@@ -12,10 +12,34 @@ import CleanCore
 protocol DashboardController: BaseController {
     func setEmail(_ email: String)
     func getEmail() -> String?
+    var cryptocurrency: Cryptocurrency? { get set }
 }
 
 final class DashboardControllerImpl: BaseControllerImpl {
-    private var email: String?    
+    private var email: String?
+    private let cryptocurrencyPriceFacade: CryptocurrencyPriceFacade
+    var cryptocurrency: Cryptocurrency? {
+        didSet {
+            notifyListenersAboutUpdate()
+        }
+    }
+    
+    init(
+        cryptocurrencyPriceFacade: CryptocurrencyPriceFacade
+    ) {
+        self.cryptocurrencyPriceFacade = cryptocurrencyPriceFacade
+        super.init()
+        Timer.scheduledTimer(withTimeInterval: 10, repeats: true, block: {_ in self.loadCryptocurrency()})
```

Comment:
```suggestion
        Timer.scheduledTimer(withTimeInterval: 10, repeats: true, block: { _ in self.loadCryptocurrency() })
```
FYI `Timer` ti bude fungovat jenom na main thread. Pokud ho z nejakyho duvodu budes inicializovat na jinem vlakne, tak ho musis zabalit do run loopu (ten ma defaultne jen main thread, tak proto).

Reply:
Tohle prosimtě pak vysvětlit prakticky. Díky